3.10.8.0 build failure

👋 trying to build the latest release, but run into some build issue. The error log is as below:


<details>
<summary>error build log</summary>

```
  /opt/homebrew/include/spdlog/fmt/fmt.h:27:14: fatal error: 'spdlog/fmt/bundled/core.h' file not found
  #    include <spdlog/fmt/bundled/core.h>
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~
  1 error generated.
```

</details>

full build log, https://github.com/Homebrew/homebrew-core/actions/runs/6589047383/job/17902765159
relates to Homebrew/homebrew-core#151928

--- a/gnuradio-runtime/include/gnuradio/io_signature.h
+++ b/gnuradio-runtime/include/gnuradio/io_signature.h
@@ -11,6 +11,9 @@
 
 #ifndef INCLUDED_IO_SIGNATURE_H
 #define INCLUDED_IO_SIGNATURE_H
+/* ensure that tweakme.h is included before the bundled spdlog/fmt header, see
+ * https://github.com/gabime/spdlog/issues/2922 */
+#include <spdlog/tweakme.h>
 
 #include <gnuradio/api.h>
 #include <spdlog/fmt/fmt.h>

--- a/gnuradio-runtime/include/gnuradio/logger.h
+++ b/gnuradio-runtime/include/gnuradio/logger.h
@@ -33,7 +33,8 @@ using logger_ptr = std::shared_ptr<void>;
 // keeps as short as possible; if anything is needed only by the implementation in
 // buffer.cc, then only include it there
 #include <gnuradio/api.h>
-#include <spdlog/common.h>
+#include <spdlog/common.h> /* important to include before spdlog/fmt/anything.h,
+                            * see https://github.com/gabime/spdlog/issues/2922 */
 #include <spdlog/fmt/fmt.h>
 #include <spdlog/fmt/ostr.h>
 #include <spdlog/version.h>

--- a/gnuradio-runtime/include/gnuradio/pmt_fmt.h
+++ b/gnuradio-runtime/include/gnuradio/pmt_fmt.h
@@ -9,6 +9,10 @@
  */
 #ifndef INCLUDED_GNURADIO_PMT_FMT_H
 #define INCLUDED_GNURADIO_PMT_FMT_H
+/* ensure that tweakme.h is included before the bundled spdlog/fmt header, see
+ * https://github.com/gabime/spdlog/issues/2922 */
+#include <spdlog/tweakme.h>
+
 #include <gnuradio/api.h>
 #include <pmt/pmt.h>
 #include <spdlog/fmt/fmt.h>

--- a/gnuradio-runtime/lib/io_signature.cc
+++ b/gnuradio-runtime/lib/io_signature.cc
@@ -10,6 +10,11 @@
  */
 
 #include <gnuradio/io_signature.h>
+
+/* ensure that tweakme.h is included before the bundled spdlog/fmt header, see
+ * https://github.com/gabime/spdlog/issues/2922 */
+#include <spdlog/tweakme.h>
+
 #include <spdlog/fmt/fmt.h>
 #include <string_view>
 #include <algorithm>

--- a/gnuradio-runtime/lib/pmt_fmt.cc
+++ b/gnuradio-runtime/lib/pmt_fmt.cc
@@ -8,6 +8,11 @@
  *
  */
 #include <gnuradio/pmt_fmt.h>
+
+/* ensure that tweakme.h is included before the bundled spdlog/fmt header, see
+ * https://github.com/gabime/spdlog/issues/2922 */
+#include <spdlog/tweakme.h>
+
 #include <pmt/pmt.h>
 #include <spdlog/fmt/fmt.h>
 #include <string_view>

--- a/gnuradio-runtime/python/gnuradio/gr/bindings/io_signature_python.cc
+++ b/gnuradio-runtime/python/gnuradio/gr/bindings/io_signature_python.cc
@@ -14,7 +14,7 @@
 /* BINDTOOL_GEN_AUTOMATIC(0)                                                       */
 /* BINDTOOL_USE_PYGCCXML(0)                                                        */
 /* BINDTOOL_HEADER_FILE(io_signature.h)                                        */
-/* BINDTOOL_HEADER_FILE_HASH(3a49a84425fae24e08f0bfd9b4fab917)                     */
+/* BINDTOOL_HEADER_FILE_HASH(749209af8901aa05aa68915c3a8cd941)                     */
 /***********************************************************************************/
 
 #include <pybind11/complex.h>

--- a/gnuradio-runtime/python/gnuradio/gr/bindings/logger_python.cc
+++ b/gnuradio-runtime/python/gnuradio/gr/bindings/logger_python.cc
@@ -15,7 +15,7 @@
 /* BINDTOOL_GEN_AUTOMATIC(0)                                                       */
 /* BINDTOOL_USE_PYGCCXML(0)                                                        */
 /* BINDTOOL_HEADER_FILE(logger.h)                                                  */
-/* BINDTOOL_HEADER_FILE_HASH(aa1fad0b5f0d52d36816dfeb45194212)                     */
+/* BINDTOOL_HEADER_FILE_HASH(b6745a64dce4f006d5bece226c75a01e)                     */
 /***********************************************************************************/
 
 #include <pybind11/complex.h>

--- a/gr-analog/lib/agc3_cc_impl.cc
+++ b/gr-analog/lib/agc3_cc_impl.cc
@@ -10,6 +10,10 @@
 
 #include "agc3_cc_impl.h"
 
+/* ensure that tweakme.h is included before the bundled spdlog/fmt header, see
+ * https://github.com/gabime/spdlog/issues/2922 */
+#include <spdlog/tweakme.h>
+
 #include <gnuradio/io_signature.h>
 #include <spdlog/fmt/fmt.h>
 #include <volk/volk.h>

--- a/gr-blocks/lib/blockinterleaving.cc
+++ b/gr-blocks/lib/blockinterleaving.cc
@@ -9,6 +9,11 @@
  */
 
 #include <gnuradio/blocks/blockinterleaving.h>
+
+/* ensure that tweakme.h is included before the bundled spdlog/fmt header, see
+ * https://github.com/gabime/spdlog/issues/2922 */
+#include <spdlog/tweakme.h>
+
 #include <spdlog/fmt/fmt.h>
 #include <algorithm>
 #include <numeric>

--- a/gr-blocks/lib/keep_m_in_n_impl.cc
+++ b/gr-blocks/lib/keep_m_in_n_impl.cc
@@ -9,6 +9,11 @@
  */
 
 #include "keep_m_in_n_impl.h"
+
+/* ensure that tweakme.h is included before the bundled spdlog/fmt header, see
+ * https://github.com/gabime/spdlog/issues/2922 */
+#include <spdlog/tweakme.h>
+
 #include <gnuradio/io_signature.h>
 #include <spdlog/fmt/fmt.h>
 

--- a/gr-blocks/lib/message_debug_impl.cc
+++ b/gr-blocks/lib/message_debug_impl.cc
@@ -11,6 +11,11 @@
 
 #include "message_debug_impl.h"
 #include <gnuradio/io_signature.h>
+
+/* ensure that tweakme.h is included before the bundled spdlog/fmt header, see
+ * https://github.com/gabime/spdlog/issues/2922 */
+#include <spdlog/tweakme.h>
+
 #include <pmt/pmt.h>
 #include <spdlog/common.h>
 #include <spdlog/fmt/fmt.h>

--- a/gr-blocks/tests/benchmark_common.h
+++ b/gr-blocks/tests/benchmark_common.h
@@ -10,6 +10,10 @@
  */
 #ifndef INCLUDED_BENCHMARK_COMMON
 #define INCLUDED_BENCHMARK_COMMON
+/* ensure that tweakme.h is included before the bundled spdlog/fmt header, see
+ * https://github.com/gabime/spdlog/issues/2922 */
+#include <spdlog/tweakme.h>
+
 #include <spdlog/fmt/fmt.h>
 #include <string_view>
 #include <chrono>

--- a/gr-blocks/tests/benchmark_nco.cc
+++ b/gr-blocks/tests/benchmark_nco.cc
@@ -9,6 +9,10 @@
  *
  */
 
+/* ensure that tweakme.h is included before the bundled spdlog/fmt header, see
+ * https://github.com/gabime/spdlog/issues/2922 */
+#include <spdlog/tweakme.h>
+
 #include <gnuradio/fxpt_nco.h>
 #include <gnuradio/math.h>
 #include <gnuradio/nco.h>

--- a/gr-blocks/tests/benchmark_vco.cc
+++ b/gr-blocks/tests/benchmark_vco.cc
@@ -8,7 +8,13 @@
  * SPDX-License-Identifier: GPL-3.0-or-later
  *
  */
+
 #include "benchmark_common.h"
+
+/* ensure that tweakme.h is included before the bundled spdlog/fmt header, see
+ * https://github.com/gabime/spdlog/issues/2922 */
+#include <spdlog/tweakme.h>
+
 #include <gnuradio/fxpt_vco.h>
 #include <gnuradio/math.h>
 #include <spdlog/fmt/fmt.h>

--- a/gr-filter/lib/rational_resampler_impl.cc
+++ b/gr-filter/lib/rational_resampler_impl.cc
@@ -13,6 +13,11 @@
 #endif
 
 #include "rational_resampler_impl.h"
+
+/* ensure that tweakme.h is included before the bundled spdlog/fmt header, see
+ * https://github.com/gabime/spdlog/issues/2922 */
+#include <spdlog/tweakme.h>
+
 #include <gnuradio/fft/window.h>
 #include <gnuradio/filter/firdes.h>
 #include <gnuradio/io_signature.h>

